### PR TITLE
Change file descriptors threshold handling at startup

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Change file descriptors threshold handling at startup

On Linux and MacOS, require at least 8192 usable file descriptors by default to start arangod.
If less file descriptors are available to the arangod process, then the startup is automatically aborted.

Even the chosen minimum value of 8192 will often not be high enough to store considerable amounts of data. However, no higher value was chosen in order to not make too many existing installations fail after upgrading.

The required number of file descriptors can be configured using the startup option `--server.descriptors-minimum`. It defaults to 8192, but it can be increased to ensure that arangod can make use of a sufficiently high number of files. 

Setting `--server.descriptors-minimum` to a value of `0` will make the startup require only an absolute minimum limit of 1024 file descriptors, effectively disabling the change.
Such low values should only be used to bypass the file descriptors check in case of an emergency, but this is not recommended for production.

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [x] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [x] :book: CHANGELOG entry made

#### Backports:

- [x] No backports required

### Related Information

- [x] Docs PR: https://github.com/arangodb/docs/pull/613

### Testing & Verification

- [x] This change is a trivial rework / code cleanup without any test coverage.
- [x] The behavior in this PR was *manually tested*

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13785/

### CODE REVIEW

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13785/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13787/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13791/

File: CHANGELOG [hunk 1]
```diff
@@ -1,5 +1,23 @@
 devel
 -----
+
+* On Linux, require at least 8192 usable file descriptors at startup.
```
Comment (jsteemann) [inline CHANGELOG:4]:
```suggestion
* On Linux and MacOS, require at least 8192 usable file descriptors at startup. 
```

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13795/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13799/

**Comment by jsteemann** [comment]
Tests blue

**Review by mpoeter — APPROVED** [review]

**Pull request merged**